### PR TITLE
fix crash when using navigation bar from search result

### PR DESCRIPTION
**Pull request by 3nids**

### CODE REVIEW

File: src/core/featurelistextentcontroller.h [hunk 1]
```diff
@@ -35,6 +35,11 @@ class FeatureListExtentController : public QObject
     FeatureListExtentController( QObject *parent = nullptr );
     ~FeatureListExtentController();
 
+    FeatureListModelSelection *selection() const;
+    MultiFeatureListModel *model() const;
+
+    void requestFeatureFormState();
```
Comment (m-kuhn) [inline src/core/featurelistextentcontroller.h]: API docs

**Comment by qfield-fairy** [comment]
Uploaded test apks for [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-417-7d4cc1ba1f13f28561f6a35c01644f951c4cd3b7-armv7.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-417-7d4cc1ba1f13f28561f6a35c01644f951c4cd3b7-x86.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-417-c177bf269ad73b6b9592554fc3a686b3ac6d1d42-armv7.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-417-c177bf269ad73b6b9592554fc3a686b3ac6d1d42-x86.apk)

**Comment by m-kuhn** [comment]
Do you still have the stack trace around to paste on the pull request text?

**Comment by 3nids** [comment]
this solves a crash here 
image

moving the call of setting the selection and the change of state from QML to C++ fixes the crash

**Pull request merged**